### PR TITLE
DatabricksHook: fix status property to work with ClientResponse used in async mode

### LINKED ISSUE #43269: Databricks Provider fails in deferred mode due to `AttributeError: 'ClientResponse' object has no attribute 'status_code'`
### Apache Airflow Provider(s)

databricks

### Versions of Apache Airflow Providers

6.11.0

### Apache Airflow version

2.9.3

### Operating System

Kubernetes

### Deployment

Official Apache Airflow Helm Chart

### Deployment details

Repackaged of airflow image with Oracle driver

### What happened

When using the DatabricksRunNowOperator with `deferrable=True` the databricks job gets started but the Task fails at the first poll to check the job status with 

```
[2024-10-22, 16:40:13 CEST] {baseoperator.py:1695} ERROR - Trigger failed:
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/jobs/triggerer_job_runner.py", line 529, in cleanup_finished_triggers
    result = details["task"].result()
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/jobs/triggerer_job_runner.py", line 601, in run_trigger
    async for event in trigger.run():
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/databricks/triggers/databricks.py", line 88, in run
    run_state = await self.hook.a_get_run_state(self.run_id)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/databricks/hooks/databricks.py", line 450, in a_get_run_state
    response = await self._a_do_api_call(GET_RUN_ENDPOINT, json)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/databricks/hooks/databricks_base.py", line 625, in _a_do_api_call
    async for attempt in self._a_get_retry_object():
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/asyncio/__init__.py", line 166, in __anext__
    do = await self.iter(retry_state=self._retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/asyncio/__init__.py", line 153, in iter
    result = await action(retry_state)
             ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/_utils.py", line 99, in inner
    return call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 398, in <lambda>
    self._add_action_func(lambda rs: rs.outcome.result())
                                     ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 449, in result
    return self.__get_result()
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/concurrent/futures/_base.py", line 401, in __get_result
    raise self._exception
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/databricks/hooks/databricks_base.py", line 641, in _a_do_api_call
    self.log.debug("Response Status Code: %s", response.status_code)
                                               ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'ClientResponse' object has no attribute 'status_code'
```

This was working with version `6.3.0`
It works fine with `deferred=False`

### What you think should happen instead

The task should continue polling until the databricks job has finished.

### How to reproduce

Use a simple DatabricksRunNowOperator with the mentionned version above. Databricks runs on Azure.

```python
    job_run = DatabricksRunNowOperator(task_id="databricks_job", job_name="TestJob",
                                       databricks_conn_id="databricks", deferrable=True)
```

### Anything else

Happens every time

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by lucafurrer**
closes: #43269 

The object returned in by `request_func` in the `_a_do_api_call` is not a `request.Response` as in the synchronous call but a `ClientResponse` which does not have the property `status_code` but `status`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
Can you please add/modify a unit test for it to avoid regression?

**Comment by lucafurrer** [comment]
> Can you please add/modify a unit test for it to avoid regression?

Sure, I will try

**Comment by rawwar** [comment]
> > Can you please add/modify a unit test for it to avoid regression?
> 
> Sure, I will try

@lucafurrer , if you would like some help, I can write the test case for this. Thanks for fixing it.

**Comment by lucafurrer** [comment]
> > > Can you please add/modify a unit test for it to avoid regression?
> > 
> > 
> > Sure, I will try
> 
> @lucafurrer , if you would like some help, I can write the test case for this. Thanks for fixing it.

@rawwar Hi, I will not have time before tonight to look into it. If you want go for it.

**Comment by rawwar** [comment]
> @rawwar Hi, I will not have time before tonight to look into it. If you want go for it.

Won't be able to do it right now. But, i can finish this in this week

**Comment by lucafurrer** [comment]
@potiuk, @rawwar I added an unit test. Comments are welcome. 
Challenge is that the problem only appears when logging is enabled and the object which had invalid property is mocked as it is the response from the API.

File: providers/tests/databricks/hooks/test_databricks.py [hunk 1]
```diff
@@ -1309,6 +1309,20 @@ async def test_async_do_api_call_respects_schema(self, mock_get):
         mock_get.assert_called_once()
         assert mock_get.call_args.args == (f"http://{HOST}:7908/api/2.1/foo/bar",)
 
+    @pytest.mark.asyncio
```
Comment (rawwar) [inline providers/tests/databricks/hooks/test_databricks.py:1312]: Do we need to add another test? Enabling debug logging in any other async test that checks the return value will also cover this case, right?

Comment (potiuk) [reply providers/tests/databricks/hooks/test_databricks.py:1312]: Genrally speaking if you add something and you do not add or modify a test for it and the tests succeed before and after your addition - it means that the tests do not test what you added. They might accidentally work but they do not assert that what you added works as expected.

**Review by potiuk — APPROVED** [review]

File: providers/tests/databricks/hooks/test_databricks.py (hunk 1 above)
Comment (rawwar) [reply providers/tests/databricks/hooks/test_databricks.py:1312]: Thanks! I will remember to write tests that explicitly test the changes.

**Pull request merged**